protocol/src/main/java/org/fidoalliance/fdo/protocol/CryptoService.java

```diff
@@ -1206,6 +1208,26 @@ public KeyExchangeResult getSharedSecret(byte[] message, Composite ownState, Key
       case Const.ASYMKEX2048_ALG_NAME:
       case Const.ASYMKEX3072_ALG_NAME:
         return getAsymkexSharedSecret(message, ownState, decryptionKey);
+      case DiffieHellman.DH14_ALG_NAME:
+      case DiffieHellman.DH15_ALG_NAME:
+        // All owner states have our message as element 0 (FIRST_KEY) and the algorithm name
+        // in element 1 (SECOND_KEY).
+        //
+        // Restore our keyExchange from the KE state,
+        // which is element 2 (THIRD_KEY) in the owner state.
+        DiffieHellman.KeyExchange ke = new KeyExchange(ownState.getAsComposite(Const.THIRD_KEY));
+        try {
+          return new KeyExchangeResult(
```

Comment:
Where is the catch block for the above statement?

Reply:
There is no catch block, it is a try-finally guarding the 'ke' resource.  Try-finally guarantees that the finally clause will execute no matter how the block is exited.